### ISSUE TITLE
Consider aligning with W3C MiniApp Manifest

### ISSUE THREAD

**Issue by tomayac** [mikewest/purposeful-permissions#1]
The [W3C MiniApp Manifest](https://www.w3.org/TR/miniapp-manifest/) spec is a registry of supplementary members for the [Web Application Manifest](https://www.w3.org/TR/appmanifest/) and the [Web App Manifest - Application Information](https://www.w3.org/TR/manifest-app-info/) specs.

W3C MiniApp Manifest defines a [`"req_permissions"`](https://www.w3.org/TR/miniapp-manifest/#req_permissions-member) member that has an optional [`"reason"`](https://www.w3.org/TR/miniapp-manifest/#dfn-reason) property. It might make sense to align here.

```json
{
  "req_permissions": [
    {
      "name": "geolocation",
      "reason": "Find stores near you."
    }
  ]
}
```

(Note that this comes with a translation challenge, which is an unresolved Web App Manifest challenge, tracked in https://github.com/w3c/manifest/issues/676.)

**Comment by mikewest** [comment]
Combining this with the app manifest isn't at all unreasonable. That said, given the mismatch between manifest scope (paths) and capability scope (origin), it's not clear to me that we can do so in all cases.

That said, I do periodically wonder whether we could reboot [Origin Policy](https://github.com/WICG/origin-policy) through a `/.well-known/` manifest file whose scope was locked to `/`...

**Comment by tomayac** [comment]
The [`scope_extensions` proposal](https://github.com/WICG/manifest-incubations/blob/gh-pages/scope_extensions-explainer.md) complicates things further. But FWIW, I saw the scope (and the scope extensions) discussion only ever focused on navigation, but not permissions. Having permissions in a manifest is something developers are familiar with from [native app development](https://developer.android.com/guide/topics/manifest/manifest-intro#perms). On the other side, having permissions (and their justifications) in a `.well-known` file has the advantage of not requiring a full Web App Manifest, if you’re not interested in its web app aspects.

**Comment by mikewest** [comment]
> But FWIW, I saw the scope (and the scope extensions) discussion only ever focused on navigation, but not permissions.

For most app manifest use cases, that makes sense. You're either in the app or not, and navigation controls that experience.

Capability's scope does not respect app boundaries, so it makes little sense to me for one app to declare a purpose that somehow maps to the entire origin. The origin, on the other hand, should be capable of making declarations that govern each of the apps it hosts.

Native app manifests have the advantage of encompassing the entirety of the app's scope. If we want to use web app manifests for permission declaration (and again, that's not at all crazy!), I think we can do it only in cases where the declaration is scoped to the origin.

**Comment by tomayac** [comment]
> Native app manifests have the advantage of encompassing the entirety of the app's scope. If we want to use web app manifests for permission declaration (and again, that's not at all crazy!), I think we can do it only in cases where the declaration is scoped to the origin.

Many native apps extend their scope through web views. Thinking of the [tagesschau app](https://play.google.com/store/apps/details?id=de.tagesschau) (that you are probably familiar with). From the main stream app page of news, they sometimes link out to web view content hosted on [other institutions](https://en.wikipedia.org/wiki/ARD_(broadcaster)#Institutions_and_member_organizations) that gets represented inline (so most users presumably think of it as tagesschau app content, and not, say `https://www.rbb-online.de/` content). In this case, it's _not_ really a problem, because the extended scopes do not inherit the permissions from the main app scope. If anything, a user could believe that by installing tagesschau they grant permission access to all the web view origins. I reckon this is not very realistic, just wanted to make us think of web views in native apps, too.

<img width="776" alt="Screenshot 2023-09-18 at 17 32 10" src="https://github.com/mikewest/purposeful-permissions/assets/145676/ec699633-dfca-4cce-99b1-aa11f343407c">

**Comment by mikewest** [comment]
I agree with you that this is "_not_ really a problem", and honestly feels tangential to the conversation above. :) You're right that the app can load pages into a webview, but those pages only have access to the set of capabilities the app has access to and delegates to them. This is similar to frames on a page in a browser that supports permission delegation. In both cases, the app is responsible for the capability's usage and availability, and seems like the entity that ought to be making declarations.
